### PR TITLE
Add Docker CI

### PR DESCRIPTION
**Pull request by jjcollinge**
Added initial Docker CI (#1011), unified ARM container image param names and neutralised ENV var names (#1019).

This is just a starting point - it'll need integrating with the existing Jenkins CI and some discussion around how/if you want to do that is needed.

### CODE REVIEW

File: opencga-app/app/scripts/docker/Makefile [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+# `make APP_NAME=opencga-app tag` : will tag the specified app image should it exist
+# `make APP_NAME=opencga-app publish` : will push the specified app image should it exist
+
+# Override the `make_env` file default location by setting ENVFILE env var
+ENVFILE ?= make_env
+include $(ENVFILE)
+export
+
+# Check required variables are set for target build
+ifeq ($(MAKECMDGOALS), build)
```
Comment (jjcollinge) [inline opencga-app/app/scripts/docker/Makefile]: Will not work if multiple targets passed

**Comment by lawrencegripper** [comment]
@jjcollinge sounds good to me!

**Comment by jjcollinge** [comment]
Follow up issue #1052

**Comment by jjcollinge** [comment]
Follow up issue #1056 and #1058

**Comment by jjcollinge** [comment]
Associated issue #1055

**Comment by wbari** [comment]
Following is a sample jenkin pipeline to build and publish docker images : 

stage ('Build Docker Images') {
 when {
                allOf {
                    branch 'master'
                    buildingTag()
                }
             }
             steps {
                      script {
  			     def images = ["opencga", "opencga-app", "opencga-daemon", "opencga-init", "iva"]
                       for(int i =0; i < images.size(); i++){
                   sh "docker build -f opencga-app/app/scripts/docker/'${images[i]}'/Dockerfile -t opencb/opencga:$TAG_NAME ."
                   }
                 }
             }
}

 stage ('Publish Docker Images') {
             when {
                allOf {
                    branch 'master'
                    buildingTag()
                }
             }
             steps {
	       for(int i =0; i < images.size(); i++){
                      withDockerRegistry([ credentialsId: "wasim-docker-hub", url: "" ]) {
                      sh 'docker push opencb/${image[i]}:$$TAG_NAME'
                      }
                 }
             }
}

For MVP, we can leave the when conditions ( as it ll have some time Azure branch will be merged to master) and can simply build docker images in a step like this or just call @joni make file and then use Jenkings creds ID to publish them to dockerHub.  This way, we don't need recursive makefile and put any password in makefiles to publish. As there is no tag available right now in Azure, we can use commit msg as done in make file : 

                              "def tag = sh(returnStdout: true, script: "git rev-parse --verify HEAD").trim()"

stage ('Build Docker Images') {
             steps {
                      script {
 def images = ["opencga", "opencga-app", "opencga-daemon", "opencga-init", "iva"]
                              def tag = sh(returnStdout: true, script: "git rev-parse --verify HEAD").trim()
                  for(int i =0; i < images.size(); i++){
                   sh "docker build -f opencga-app/app/scripts/docker/'${images[i]}'/Dockerfile -t opencb/opencga:${tag} ."
                   }
                 }
                 }
             }
@jjcollinge please have a look if these help.

**Comment by jjcollinge** [comment]
@wbari thanks for this, let's discuss this further on #1056 so that it doesn't block this PR.

File: opencga-app/app/scripts/docker/Makefile.docker [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+# Builds and tags the container
+.PHONY: build
+build:
+	docker build $(DOCKER_BUILD_ARGS) -t $(APP_NAME) -f $(PATH_PREFIX)/Dockerfile .
+
+# Tags a docker image with git commit SHA
+.PHONY: tag
+tag:
+	docker tag $(APP_NAME) $(DOCKER_SERVER)$(DOCKER_REPO)/$(APP_NAME):$(TAG)
```
Comment (wbari) [inline opencga-app/app/scripts/docker/Makefile.docker]: @joni Just for naming consistency, "DOCKER_REPO" is called organisation "opencb" in DockerHub , we should rename this to DOCKER_ORGANIZATION.

File: opencga-app/app/scripts/docker/Makefile.docker [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+# Logs in to a docker registry
+.PHONY: login
+login:
+	@echo $(DOCKER_PASSWORD) | docker login -u $(DOCKER_USERNAME) --password-stdin $(DOCKER_SERVER)
+
+# Publishes a docker image to docker registry
+.PHONY: publish
+publish: tag login
+	docker push $(DOCKER_SERVER)$(DOCKER_REPO)/$(APP_NAME):$(TAG)
```
Comment (wbari) [inline opencga-app/app/scripts/docker/Makefile.docker]: same here as above
docker push opencb/opencga-daemon:tagname

**Comment by wbari** [comment]
For some reasons, these script from PR 1042 not working for me,  while running the following command from : 
~/appl/opencga/opencga-app/app/scripts/docker[PR-1042]$ make DOCKER_USERNAME="myUserName" DOCKER_PASSWORD="myPassword" PUBLISH=true TAG="v0.0.1" DOCKER_REPO="opencb" throws the following error: 

image

File: opencga-app/app/scripts/docker/Makefile.docker (hunk 2 above)
Comment (wbari) [reply opencga-app/app/scripts/docker/Makefile.docker]: Here App name is basically Repository name. 
image

Comment (jjcollinge) [reply opencga-app/app/scripts/docker/Makefile.docker]: @wbari thanks, yeh I was going with Azure terminology originally but it makes more sense to move to DH if that's your expected registry. I've updated this now.

**Comment by jjcollinge** [comment]
@wbari you are absolutely correct, I must of missed this in an earlier merge - fixing now - new commit in 2 mins. Sorry about that.

**Pull request merged**